goal_src/jak3/engine/collide/collide-cache.gc

```diff
@@ -7,3 +7,898 @@
+      (if (< (fabs (-> arg0 move-dist y)) 4096.0)
+          (+! v1-3 1)
+          )
+      (if (< (fabs (-> arg0 move-dist z)) 4096.0)
+          (+! v1-3 1)
+          )
+      (when (< 1 v1-3)
+        (set-from-point-offset-pad! (-> arg0 bbox) (-> arg0 start-pos) (-> arg0 move-dist) (-> arg0 radius))
+        (fill-using-bounding-box this arg0)
+        (b! #t cfg-55 :delay (nop!))
+        (the-as none 0)
+        )
+      )
+    (nop!)
+    (let ((v1-10 (-> arg0 ignore-pat-s32)))
+      (nop!)
+      (let ((a0-14 (-> arg0 ignore-process0)))
+        (nop!)
+        (let ((a1-3 (-> arg0 ignore-process1)))
+          (nop!)
+          (let ((a2-1 (-> arg0 collide-with-s32)))
+            (nop!)
+            (.lvf vf9 (&-> arg0 exit-planes 0 quad))
+            (nop!)
+            (.lvf vf3 (&-> arg0 move-dist quad))
+            (nop!)
+            (.mov.vf vf13 vf0)
+            (nop!)
+            (.lvf vf1 (&-> arg0 start-pos quad))
+            (.mul.vf vf8 vf3 vf3)
+            (nop!)
+            (.add.vf vf2 vf1 vf3)
+            (set! (-> this ignore-mask) (the-as pat-surface v1-10))
+            (.add.y.vf vf8 vf8 vf8 :mask #b1)
+            (set! (-> this num-tris) 0)
+            (.min.vf vf4 vf1 vf2)
+            (set! (-> this num-prims) 0)
+            (.max.vf vf5 vf1 vf2)
+            (set! (-> this collide-with) (the-as collide-spec a2-1))
+            )
+          (.sub.w.vf vf10 vf0 vf9 :mask #b111)
+          (set! (-> this ignore-processes 0) (the-as process a0-14))
+          (.add.z.vf vf8 vf8 vf8 :mask #b1)
+          (set! (-> this ignore-processes 1) (the-as process a1-3))
+          )
+        )
+      )
+    (.sub.w.vf vf4 vf4 vf9 :mask #b111)
+    (nop!)
+    (.add.w.vf vf5 vf5 vf9 :mask #b111)
+    (nop!)
+    (.ftoi.vf vf15 vf10)
+    (nop!)
+    (.isqrt.vf Q vf0 vf8 :fsf #b11 :ftf #b0)
+    (nop!)
+    (.add.w.vf vf11 vf0 vf9 :mask #b111)
+    (nop!)
+    (nop!)
+    (nop!)
+    (nop!)
+    (.svf (&-> arg0 local-box4w min quad) vf15)
+    (.ftoi.vf vf6 vf4)
+    (.svf (&-> this collide-box min quad) vf4)
+    (.ftoi.vf vf7 vf5)
+    (.svf (&-> this collide-box max quad) vf5)
+    (nop!)
+    (.svf (&-> arg0 bbox min quad) vf4)
+    (nop!)
+    (.svf (&-> arg0 bbox max quad) vf5)
+    (nop!)
+    (.svf (&-> this collide-box4w min quad) vf6)
+    (nop!)
+    (.svf (&-> this collide-box4w max quad) vf7)
+    (nop!)
+    (.svf (&-> arg0 bbox4w min quad) vf6)
+    (nop!)
+    (.svf (&-> arg0 bbox4w max quad) vf7)
+    (.wait.vf)
+    (nop!)
+    (.add.vf vf8 vf0 Q :mask #b1)
+    (nop!)
+    (.mul.x.vf vf12 vf3 vf8)
+    (nop!)
+    (.nop.vf)
+    (nop!)
+    (.div.vf Q vf0 vf8 :fsf #b11 :ftf #b0)
+    (nop!)
+    (.mul.vf vf22 vf12 vf12)
+    (nop!)
+    (.abs.vf vf23 vf12)
+    (nop!)
+    (.add.y.vf vf22 vf22 vf22 :mask #b1)
+    (.mov v1-11 vf23)
+    (.wait.vf)
+    (nop!)
+    (.add.vf vf8 vf0 Q :mask #b1)
+    (nop!)
+    (b! (zero? v1-11) cfg-12 :likely-delay (.add.z.vf vf13 vf0 vf12 :mask #b1))
```

Comment:
This will check if the lower 64-bits of `v1-11` in the original game (I think - you should check asm or jak 2 version of this), so this needs to be a 64-bit register and use `int`/`uint` type.